docs/topics/tasks.txt

```diff
@@ -0,0 +1,418 @@
+These arguments are provided as additional keys in the :setting:`TASKS` setting.
+Valid arguments are as follows:
+
+* :setting:`ENQUEUE_ON_COMMIT <TASKS-ENQUEUE_ON_COMMIT>`: Whether a Task should
+  be enqueued at the end of the current transaction (if there
+  is one) commits successfully, rather than enqueueing immediately. This argument
+  defaults to ``True``.
+
+* :setting:`QUEUES <TASKS-QUEUES>`: Restrict the queue names a Task may be
+  enqueued to. By default, only the ``"default"`` queue is allowed. Queue name
+  validation can be disabled by setting this to an empty list.
+
+* :setting:`OPTIONS <TASKS-OPTIONS>`: Extra parameters can be passed to the
+  task backend. Available parameters vary depending on the task backend.
+
+Asynchronous support
+--------------------
+
+Django has developing support for asynchronous Task backends.
+
+``django.tasks.backends.base.BaseTaskBackend`` has async variants of all base
+methods. By convention, the asynchronous versions of all methods are prefixed
+with ``a``. The arguments for both variants are the same.
+
+Retrieving backends
+-------------------
+
+Backends can be retrieved using the ``tasks`` connection handler::
+
+    from django.tasks import tasks
+
+    tasks["default"]  # The default backend
+    tasks["reserve"]  # Another backend
+
+The "default" backend is available as ``default_task_backend``::
+
+    from django.tasks import default_task_backend
+
+Backend features
+----------------
+
+Django's tasks framework is designed to be backend-agnostic. This means some
+backends may not implement the same features as others, such as executing ``async``
+functions.
+
+Depending on the backend configured, it may be necessary to gracefully degrade
+functionality, or use a system check to prevent the application from starting at all.
+
+To facilitate this, certain features can be checked on a backend:
+
+* :attr:`supports_defer <django.tasks.backends.base.BaseTaskBackend.supports_defer>`:
+  Can Tasks be executed after a specific time using ``run_after``?
+
+* :attr:`supports_async_task <django.tasks.backends.base.BaseTaskBackend.supports_async_task>`:
+  Can ``async`` functions (coroutines) be used as task functions?
+
+* :attr:`supports_get_result <django.tasks.backends.base.BaseTaskBackend.supports_get_result>`:
+  Can a Task's results be retrieved from another thread or process?
+
+Defining tasks
+==============
+
+Tasks are defined using the :meth:`django.tasks.task` decorator on a
+module-level function::
+
+    from django.core.mail import send_mail
+    from django.tasks import task
+
+
+    @task
+    def email_users(emails, subject, message):
+        return send_mail(
+            subject=subject, message=message, from_email=None, recipient_list=emails
+        )
+
+
+The return value of the decorator is a :class:`django.tasks.Task` instance.
+
+.. note::
+
+    All arguments and return values must be JSON serializable, and round-trip
+    as the same types.
```

Comment:
I got confused by "and round-trip as the same type." - what is this referring to sorry?

Reply:
Yeah I really struggled finding words for this. Basically, if you serialize it to JSON and bring it back, it should be the same type. For example, `json.dumps((1,2,3))` returns `"[1,2,3]"`, which is a list, not a tuple, so the type is lost. Because there's no separate schema, the types must be identical when converted to and from JSON.

Reply:
I pushed a tweak :+1: